src/las_io.cpp

```diff
@@ -264,9 +263,15 @@ bool PointArray::loadLas(QString fileName, size_t maxPointCount,
         *numReturns++ = point.number_of_returns_of_given_pulse;
 #       endif
         *pointSourceId++ = point.point_source_ID;
-        // Put flags back in classification byte to avoid memory bloat
-        *classification++ = point.classification | (point.synthetic_flag << 5) |
-                            (point.keypoint_flag << 6) | (point.withheld_flag << 7);
+
+        if (point.extended_classification) {
```

Comment:
Seems like we ought to be checking for `point.extended_point_type` here, rather than merely non-zero extended classification?

Reply:
Good point. What do you recommend?